### PR TITLE
cd: add packages deploy on push with tag

### PR DESCRIPTION
**Pull request by YozhikVolkoff**
Adds "Deploy pakages" step to .github/workflows/packaging.yml which
deploys packages to our S3-based repositories. This step is triggered
only on push with any tag. Based on [1].
Adds "Fetch tags" step before creating packages with packpack. Using
Github checkout Actions and 'git describe ..' calls w/o '--tags' flag
we are unable to get the needed tag for packages version setup.
'git fetch --tags -f' fixes it. For more information, see [2].

[1] [.github/actions/upload/action.yml from tarantool/rws](https://github.com/tarantool/rws/blob/15f2450c3826794e65f5e6e0e16694806685c451/.github/actions/upload/action.yml)
[2] [.github/actions/environment/action.yml from tarantool/tarantool](https://github.com/tarantool/tarantool/blob/3a98e0794e06bee9ecccbe2bc875e87f612d5560/.github/actions/environment/action.yml)

Needed for https://github.com/tarantool/tarantool-c/issues/145.

### CODE REVIEW

**Comment by Totktonada** [comment]
Added `RWS_AUTH` into secrets.

File: .github/workflows/packaging.yml [hunk 1]
```diff
@@ -37,7 +37,7 @@ jobs:
       DIST: ${{ matrix.platform.dist }}
 
     steps:
-      - name: Clone the module
+      - name: Clone the connector
```
Comment (LeonidVas) [inline .github/workflows/packaging.yml:40]: Not related with the patch.
https://github.com/tarantool/tarantool/wiki/Code-review-procedure#general-coding-points-to-check
`Don't do a change if it is not related to the patch...`

File: .github/workflows/packaging.yml [hunk 2]
```diff
@@ -51,3 +51,22 @@ jobs:
 
       - name: Create packages
         run: ./packpack/packpack
+
+      - name: Deploy packages
+        # We need this step to run only on push with tag
```
Comment (LeonidVas) [inline .github/workflows/packaging.yml:56]: Extra space at the end.
Set a dot at the end of the sentence.
https://github.com/tarantool/tarantool/wiki/Code-review-procedure#code-style
`Start sentences from a capital letter, end with a dot...`

File: .github/workflows/packaging.yml [hunk 3]
```diff
@@ -51,3 +51,22 @@ jobs:
...
+          CURL_CMD="curl -LfsS \
+            -X PUT ${RWS_URL_PART}/${OS}/${DIST} \
+            -u ${RWS_AUTH} \
+            -F product=${PRODUCT_NAME}"
+
+          for f in $(ls -I '*build*' -I '*.changes' ./build); do
+            CURL_CMD+=" -F $(basename ${f})=@./build/${f}"
+          done
+
+          echo ${CURL_CMD}
```
Comment (LeonidVas) [inline .github/workflows/packaging.yml:72]: It's a good practice to print out the command, but that's not enough, you still need to run it for execution).
Add yet one line like `${CURL_CMD}`.

**Review by LeonidVas — CHANGES_REQUESTED** [review]
Hi! Thank you for the patch.
As far as I can see, `ci` writes using lower case in commit titles. So, I suggest to replace `CD` with `cd`. Also https://github.com/tarantool/tarantool/wiki/Code-review-procedure#commit-message `Description is what the patch does, started from lowercase letter...`
About `Based on [1].` I don't think it's necessary. But if you want to keep this information, please get the link to https://github.com/tarantool/rws/blob/15f2450c3826794e65f5e6e0e16694806685c451/.github/actions/upload/action.yml and move it to the line before `Needed for ...`.

File: .github/workflows/packaging.yml (hunk 2 above)
Comment (YozhikVolkoff) [reply .github/workflows/packaging.yml]: Fixed.

File: .github/workflows/packaging.yml (hunk 1 above)
Comment (YozhikVolkoff) [reply .github/workflows/packaging.yml:40]: I discussed this. Now this change is added to the second commit.

File: .github/workflows/packaging.yml [hunk 4]
```diff
@@ -51,3 +51,24 @@ jobs:
 
       - name: Create packages
         run: ./packpack/packpack
+
+      - name: Deploy packages
+        # We need this step to run only on push with tag.
+        if: ${{ github.event_name == 'push' && github.event.ref_type == 'tag' }}
```
Comment (LeonidVas) [inline .github/workflows/packaging.yml:57]: `ref_type` present only in  `github.event_name == 'create'`  https://docs.github.com/en/developers/webhooks-and-events/events/github-event-types

**Review by LeonidVas — CHANGES_REQUESTED** [review]

**Comment by Totktonada** [comment]
I expect the following:

1. A confirmation that the tag push condition is working. I suggest to just use the [well known solution](https://github.com/tarantool/expirationd/blob/838c2d16346741cd753f6a95810284ecb4c6b50a/.github/workflows/publish.yml#L20).
2. Workaround for the tag deannotation problem https://github.com/actions/checkout/issues/290. See the [known solution](https://github.com/tarantool/tarantool/blob/3a98e0794e06bee9ecccbe2bc875e87f612d5560/.github/actions/environment/action.yml#L50-L72). (Don't leave just one command: it is hacky and should be commented.)

File: .github/workflows/packaging.yml (hunk 4 above)
Comment (YozhikVolkoff) [reply .github/workflows/packaging.yml]: I changed the condition to `startsWith(github.ref, 'refs/tags/')`.

Comment (LeonidVas) [reply .github/workflows/packaging.yml]: It seems to work fine.

**Review by LeonidVas — APPROVED** [review]

**Review by Totktonada — APPROVED** [review]
Thanks for the patch!

LGTM. At least visually it looks that it should do the job. I'm okay to go ahead and try.

I would close the issue from the commit message, it seems completely resolved.

**Review by Korablev77 — APPROVED** [review]
> ... our S3-based repositiries
repositories :)

LGTM
